lib/topics_filter.rb

```diff
@@ -695,15 +755,14 @@ def include_topics_with_any_tags(tag_ids)
       column: "topics.views",
     },
     "read" => {
-      column: "tu1.last_visited_at",
+      column: "tu.last_visited_at",
       scope: -> do
-        if @guardian.user
-          @scope.joins(
-            "JOIN topic_users tu1 ON tu1.topic_id = topics.id AND tu1.user_id = #{@guardian.user.id.to_i}",
-          ).where("tu1.last_visited_at IS NOT NULL")
+        if @guardian&.authenticated?
```

Comment:
```suggestion
        if @guardian.authenticated?
```

No need for save nav, you are always making sure you have a guardian

Reply:
true, was a last minute change